pywbem/_cim_types.py

```diff
@@ -655,11 +658,20 @@ def get_local_utcoffset():
         A positive value indicates minutes east of UTC, and a negative
         value indicates minutes west of UTC.
         """
-        utc = datetime.utcnow()
+        # Effective Python 3.12, utcnow() generates deprecation warning
+        # because utcnow returns a niave datetime object. timezone does
+        # not exist for python 2.7
+        if six.PY3:
+            aware_utc = datetime.now(timezone.utc)
+        else:
+            unaware_utc = datetime.utcnow()
+            aware_utc = unaware_utc.replace(tzinfo=timezone.utc)
+
         local = datetime.now()
-        if local < utc:
-            return - int(float((utc - local).seconds) / 60 + .5)
-        return int(float((local - utc).seconds) / 60 + .5)
+        local_unaware = local.replace(tcinfo=None)
+        if local_unaware < aware_utc:
```

Comment:
This compares naive and unaware, which will result in TypeError.

Reply:
fixed